tests/sqlparser_common.rs

```diff
@@ -5071,7 +5071,9 @@ fn parse_interval_dont_require_unit() {
 
 #[test]
 fn parse_interval_require_unit() {
-    let dialects = all_dialects_where(|d| d.require_interval_qualifier());
+    let dialects = all_dialects_where(|d| {
+        d.require_interval_qualifier() && d.is_reserved_for_identifier(Keyword::INTERVAL)
```

Comment:
oh I would have expected that these the reserved condition is always true given the require_interval condition is true? in that, would it be correct for a dialect to not have INTERVAL as a keyword yet `d.require_interval_qualifier()` returns true for that dialect